e2e/specs/translation-builder-workflows.spec.ts

```diff
@@ -0,0 +1,437 @@
+/* eslint-disable testing-library/prefer-screen-queries */
+
+import { test } from '../core';
+import { expect } from '@playwright/test';
+import { deleteForm, createForm, createValueReference, addFormResources } from '../commands/form-operations';
+import { FormBuilderPage } from '../pages';
+
+let formUuid = '';
+
+const formDetails = {
+  name: 'Covid-19 Screening',
+  description: 'A test form for recording COVID-19 screening information',
+  version: '1.0',
+  published: true,
+};
+
+test.describe('Translation Builder Workflows', () => {
+  test.beforeEach(async ({ api }) => {
+    const form = await createForm(api, false, formDetails);
+    formUuid = form.uuid;
+
+    const valueReference = await createValueReference(api);
+    await addFormResources(api, valueReference, formUuid);
+  });
+
+  test.afterEach(async ({ api }) => {
+    if (formUuid) {
+      await deleteForm(api, formUuid);
+    }
+  });
+
+  test('Workflow 1: Basic Translation Builder Display', async ({ page }) => {
+    const formBuilderPage = new FormBuilderPage(page);
+
+    await test.step('When I visit the form builder', async () => {
+      await formBuilderPage.gotoFormBuilder();
+    });
+
+    await test.step('And I search for the form and open it for editing', async () => {
+      await formBuilderPage.searchForForm(formDetails.name);
+      await formBuilderPage.page.getByRole('row', { name: formDetails.name }).getByLabel('Edit Schema').first().click();
+    });
+
+    await test.step('And I open the Translation builder tab', async () => {
+      await formBuilderPage.translationBuilderTab().click();
+    });
+
+    await test.step('Then I should see the translation builder components (Download/Upload buttons, Translation filter tabs: All/Translated/Untranslated, and the search input)', async () => {
+      const translationPanel = formBuilderPage.page
+        .getByRole('tabpanel')
+        .filter({ has: formBuilderPage.page.getByRole('button', { name: /upload translation/i }) });
+
+      await expect(translationPanel).toBeVisible();
+      await expect(translationPanel.getByRole('button', { name: /download translation/i })).toBeVisible();
+      await expect(translationPanel.getByRole('button', { name: /upload translation/i })).toBeVisible();
+      const tabList = translationPanel.getByRole('tablist', { name: 'Translation filter' });
+      await expect(tabList).toBeVisible();
+
+      const tabs = tabList.getByRole('tab');
+      await expect(tabs).toHaveCount(3);
+      await expect(tabs.nth(0)).toContainText(/all/i);
+      await expect(tabs.nth(1)).toContainText(/translated/i);
+      await expect(tabs.nth(2)).toContainText(/untranslated/i);
+      await expect(translationPanel.getByPlaceholder(/search translation keys/i)).toBeVisible();
+    });
+  });
```

Comment:
The e2e test is to test possible user interaction workflows, so testing if the UI elements are there is not something that needs a specific e2e test
```suggestion
```